src/oatpp/network/Server.cpp

```diff
@@ -50,11 +50,13 @@ void Server::conditionalMainLoop() {
 
   while (getStatus() == STATUS_RUNNING) {
     if (m_condition()) {
+      m_spinlock.lock();
       auto connection = m_connectionProvider->get();
```

Comment:
`ConnectionProvider::get()` can potentially throw.
It's better to use lock guard.